subsys/net/conn_mgr/conn_mgr_connectivity.c

```diff
@@ -263,6 +270,59 @@ static void conn_mgr_conn_handle_iface_admin_up(struct net_if *iface)
 	}
 }
 
+/**
+ * @brief Perform automated behaviors in response to any iface that loses
+ * connection and does not expect to regain it (referred to, here, as hard-down).
+ *
+ * This is how conn_mgr_conn automatically takes such ifaces admin-down.
+ *
+ * @param iface - The iface which experienced hard-down.
+ */
+static void conn_mgr_conn_handle_iface_hard_down(struct net_if *iface)
```

Comment:
Maybe just stick to the `admin_down` nomenclature?

Reply:
You are right, I could name the function after what it does, rather than the thing it responds to. Would `conn_mgr_conn_if_auto_admin_down` be an acceptable name in your eyes?